pkg/probeupload/server_handler.go

```diff
@@ -40,18 +42,36 @@ func LogCallback(logger logging.Logger) func(error) {
 	}
 }
 
-// NewProbeServerHandler returns an http.Handler for serving kernel probes. The handler assumes the path of kernel
-// probes is rooted at `/`, i.e., wrap this via `http.StripPrefix` when serving on a sub-path.
-// The errorCallback is invoked for errors that happen during writing the response body, and thus cannot be transmitted
-// to the client via status/headers. It may be nil, in which case errors are simply ignored.
-func NewProbeServerHandler(errorCallback func(error), sources ...ProbeSource) http.Handler {
-	return &handler{
+// NewProbeServerHandler returns a http.Handler for serving kernel probes.
+// It runs in Online mode by default and is meant to be used in Central.
+// The probeServerHandler assumes the path of kernel probes is rooted at `/`, i.e., wrap this via `http.StripPrefix`
+// when serving on a sub-path. The errorCallback is invoked for errors that happen during writing the response body,
+// and thus cannot be transmitted  to the client via status/headers. It may be nil, in which case errors are ignored.
+func NewProbeServerHandler(errorCallback func(error), sources ...ProbeSource) *probeServerHandler {
+	psh := NewSensorProbeServerHandler(errorCallback, sources...)
+	psh.GoOnline() // this hadler is used in Central as well and it must be immediately online for backwards compat.
```

Comment:
```suggestion
	psh.GoOnline() // this handler is used in Central as well and it must be immediately online for backwards compat.
```

Reply:
Thanks, I will add this once we have first CI results - I don't want to interrupt them now.